source/common/formatter/stream_info_formatter.cc

```diff
@@ -1603,6 +1604,15 @@ const StreamInfoFormatterProviderLookupTable& getKnownStreamInfoFormatterProvide
                     return result;
                   });
             }}},
+          {"UNIQUE_ID",
+           {CommandSyntaxChecker::COMMAND_ONLY,
+            [](const std::string&, const absl::optional<size_t>&) {
+              return std::make_unique<StreamInfoStringFieldExtractor>(
+                  [](const StreamInfo::StreamInfo&) -> absl::optional<std::string> {
+                    static Random::RandomGeneratorImpl random_generator;
+                    return absl::make_optional<std::string>(random_generator.uuid());
```

Comment:
Hi @alyssawilk - Please can you provide your input on whether the current usage of RandomeGeneratorImpl is ok to proceed with in PR?

Reply:
do we allow global static classes? I thought there were initializer concerns.  
If I'm not misremembering we could have the impl have a static function since as you say it's stateless and use that as a workaround

Reply:
Like this way: https://github.com/envoyproxy/envoy/pull/34919
cc @alyssawilk

Reply:
Yep, LGTM

Reply:
Resolved - now using static method